### ISSUE TITLE
Discovery: Update CMS edit form to create separate heading fields

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/va.gov-cms#12128]
## Description
In an effort to lessen the number of heading violations that editors are creating on their pages, an option has been suggested that the content edit form could be updated to remove/change full rich text editor fields to limited rich text fields (editors cannot add headings in here) and add heading blocks. The idea is that these new heading blocks would allow the editors to put only headings in here and would break up their content to be more easily scannable and allow us to throw a flag before publishing if a skipped heading violation was found.

How might we create a better structure of the content experience to have a better heading outline. 

### Things to consider

- Can this be done? What is the tech lift on this change?
- How would the change management be handled?
- We should do an audit of content types that have full rich text editors first to determine where these violations could occur and where this solution could eventually be applicable. (This was done #12811)
- What content type could be a good pilot?
  - Stories could be a good candidate because it has a full rich text editor but also there has been an ask to have multiple images.  Therefore, we can make this change in conjunction with a change to images and give editors something they want while also getting something we want
  - Resources & Support was also mentioned during our discussion with Dave C as a possible candidate for pilot
  - New VA Benefits may be the pilot after most recent discussions

## Acceptance Criteria
- [ ] Propose a solution and document the pros/cons of a migration
- [ ] Determine the LOE for a migration
- [ ] Determine pilot content type (R&S, stories, events???) and create recommendation (include the why)

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [x] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by EWashb** [comment]
The immediate thought is that the FE tech lift would be more intense than the BE. 

But is migration feasible? We do have a large amount of content. 

 From an editorial standpoint there may need to be some research. What are the appropriate chunks of information? Do editors really understand the differences between headings?

**Comment by EWashb** [comment]
Next possible steps from prior ACs: 
Change management plan and determine pilot content type or research process

**Comment by EWashb** [comment]
The recommendation should encompass an understanding of the veteran use and what the better value would be. Non-sitewide content editors should be in the pilot.

**Comment by EWashb** [comment]
Hypothesis: Amount of effort is decreased for the content team to validate what's needed to get to publication.

**Comment by laflannery** [comment]
I just thought of something I would like to make sure we consider here - when I first started, I sat in on a meeting with Sitewide Content and she was asking about the IDs the CMS auto assigns to headings, because this team sometimes uses these to create anchor links. However, if the text of the heading is changed, the ID will change - breaking the link.

When headings are in a rich text editor, this is not a huge deal, we can put an ID in the source code so the auto ID doesn't get created and the link won't break (this causes other issues like duplicate IDs) but what I want to keep in mind is, that now we are removing the flexibility of headings for a good reason but are we also going to cause more of these broken anchor type links for the Sitewide content team, and possibly other teams/editors?

**Comment by EWashb** [comment]
pull this into refinement during sprint 77. What are we trying to achieve here and what is the vision for the CMS?

**Comment by davidmpickett** [comment]
Some work on the Public Websites team might be an early place to experiment with this: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/13143

**Comment by laflannery** [comment]
@EWashb Since PW has done this MVP of the magic headers, should we close this? I feel like the next step here is to see what the feedback/results are with magic header and then go from there (either back to the drawing board if it goes terribly or iterate on what they have)

**Comment by EWashb** [comment]
@laflannery i agree
